modular_skyrat/modules/fishing/code/fishing.dm

```diff
@@ -69,27 +74,52 @@ GLOBAL_LIST_INIT(fishing_weights, list(
 		QDEL_NULL(mutate_parent)
 	if(!fisher)
 		return
+	var/obj/item/fishing_rod/fisher_rod = fisher
 	if(COOLDOWN_FINISHED(src, start_fishing_window) && !COOLDOWN_FINISHED(src, stop_fishing_window))
 		var/turf/fisher_turf = get_turf(fisher)
-		create_reward(fisher_turf)
+		create_reward(fisher_turf, fisher_rod.fishing_focus)
 		if(master_involved)
-			create_reward(fisher_turf)
+			create_reward(fisher_turf, fisher_rod.fishing_focus)
 
-/datum/component/fishing/proc/create_reward(turf/spawning_turf)
+/datum/component/fishing/proc/create_reward(turf/spawning_turf, focus)
 	var/atom/spawning_reward
-	switch(rand(1, 100))
-		if(1 to 50)
-			spawning_reward = pick_weight(GLOB.trash_loot)
-			while(islist(spawning_reward))
-				spawning_reward = pick_weight(spawning_reward)
-		if(51 to 75)
-			if(generate_fish)
-				generate_fish(spawning_turf, random_fish_type())
-		if(76 to 95)
-			spawning_reward = pick_weight(possible_loot)
-		if(96 to 100)
-			spawning_reward = /obj/item/skillchip/fishing_master
-	new spawning_reward(spawning_turf)
+	switch(focus)
+		if(FOCUS_NONE)
+			switch(rand(1, 100))
+				if(1 to 33)
+					spawning_reward = pick_weight(GLOB.trash_loot)
+					while(islist(spawning_reward))
+						spawning_reward = pick_weight(spawning_reward)
+				if(34 to 66)
+					if(generate_fish)
+						generate_fish(spawning_turf, random_fish_type())
+				if(67 to 98)
+					spawning_reward = pick_weight(possible_loot)
+				if(99 to 100)
+					spawning_reward = /obj/item/skillchip/fishing_master
+		if(FOCUS_TRASH)
+			switch(rand(1, 100))
+				if(1 to 98)
+					spawning_reward = pick_weight(GLOB.trash_loot)
+					while(islist(spawning_reward))
+						spawning_reward = pick_weight(spawning_reward)
+				if(99 to 100)
+					spawning_reward = /obj/item/skillchip/fishing_master
+		if(FOCUS_FISH)
+			switch(rand(1, 100))
+				if(1 to 98)
+					if(generate_fish)
+						generate_fish(spawning_turf, random_fish_type())
+				if(99 to 100)
+					spawning_reward = /obj/item/skillchip/fishing_master
+		if(FOCUS_ORE)
+			switch(rand(1, 100))
+				if(1 to 98)
+					spawning_reward = pick_weight(possible_loot)
+				if(99 to 100)
+					spawning_reward = /obj/item/skillchip/fishing_master
```

Comment:
I kinda feel like those could be procs, like, `create_reward_normal()`, `create_reward_trash()`, `create_reward_fish()` and `create_reward_ore()`, so you can return a `spawning_reward` value with them, and to keep one functionality per proc, making this overall cleaner.